patch_api/api/estimates_api.py

```diff
@@ -1017,8 +1059,15 @@ def retrieve_estimate_with_http_info(self, id, **kwargs):  # noqa: E501
             path_params["id"] = local_var_params["id"]  # noqa: E501
 
         query_params = []
+
+        # do not add duplicate keys to query_params list
+        existing_keys = []
+        for param in query_params:
+            existing_keys.append(param[0])
+
         for key in kwargs:
-            query_params.append([key, kwargs.get(key)])
+            if key not in existing_keys:
+                query_params.append([key, kwargs.get(key)])
```

Comment:
We'll probably want to address the fact that this is duplicated several times over?

Reply:
I suppose but this logic does seem useful to have in each case so there aren't duplicate params in the list